Upgrade pyarrow to >= 14.0.1

## Description

The `pyarrow==8.0.0` dependency is not compatible with python `3.11` and is causing issues in updated agents installing the AWS module dependencies. We should upgrade it to `14.0.1` or higher.

> [!Note]
> This issue originally indicated to upgrade pyarrow to `>=10.0.1`, but a [new vulnerability](https://github.com/wazuh/wazuh/security/dependabot/29) ([CVE](https://nvd.nist.gov/vuln/detail/CVE-2023-47248)) has been found in the versions prior to `14.0.1`.

```console
root@ip-172-31-38-194:/home/admin# python3 --version
Python 3.11.2
root@ip-172-31-38-194:/home/admin# pip3 --version
pip 23.2.1 from /usr/local/lib/python3.11/dist-packages/pip (python 3.11)
root@ip-172-31-38-194:/home/admin# pip3 install pyarrow==8.0.0 --break-system-packages
...
Collecting pyarrow==8.0.0
  Downloading pyarrow-8.0.0.tar.gz (846 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 846.6/846.6 kB 54.5 MB/s eta 0:00:00
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Collecting numpy>=1.16.6 (from pyarrow==8.0.0)
  Obtaining dependency information for numpy>=1.16.6 from https://files.pythonhosted.org/packages/32/6a/65dbc57a89078af9ff8bfcd4c0761a50172d90192eaeb1b6f56e5fbf1c3d/numpy-1.25.2-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata
  Downloading numpy-1.25.2-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (5.6 kB)
Downloading numpy-1.25.2-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (18.2 MB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 18.2/18.2 MB 34.9 MB/s eta 0:00:00
Building wheels for collected packages: pyarrow
  Building wheel for pyarrow (pyproject.toml) ... error
  error: subprocess-exited-with-error

  × Building wheel for pyarrow (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [536 lines of output]
  ...
  
-- Running cmake for pyarrow
      cmake -DPYTHON_EXECUTABLE=/usr/bin/python3 -DPython3_EXECUTABLE=/usr/bin/python3 "" -DPYARROW_BUILD_CUDA=off -DPYARROW_BUILD_FLIGHT=off -DPYARROW_BUILD_GANDIVA=off -DPYARROW_BUILD_DATASET=off -DPYARROW_BUILD_ORC=off -DPYARROW_BUILD_PARQUET=off -DPYARROW_BUILD_PARQUET_ENCRYPTION=off -DPYARROW_BUILD_PLASMA=off -DPYARROW_BUILD_S3=off -DPYARROW_BUILD_HDFS=off -DPYARROW_USE_TENSORFLOW=off -DPYARROW_BUNDLE_ARROW_CPP=off -DPYARROW_BUNDLE_BOOST=off -DPYARROW_GENERATE_COVERAGE=off -DPYARROW_BOOST_USE_SHARED=on -DPYARROW_PARQUET_USE_SHARED=on -DCMAKE_BUILD_TYPE=release /tmp/pip-install-rp51_v2o/pyarrow_e1a36e92f19d475fa922148f8cc544b5
      error: command 'cmake' failed: No such file or directory
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for pyarrow
Failed to build pyarrow
ERROR: Could not build wheels for pyarrow, which is required to install pyproject.toml-based projects
```

Related issue: https://github.com/wazuh/wazuh-documentation/issues/6470
Python 3.11 pyarrow support addition: https://github.com/apache/arrow/pull/14499

## Checks
<!-- Do not modify, this will be ticked during development -->
The following elements have been updated or reviewed (should also be checked if no modification is required):
- [x] Tests (unit tests, API integration tests).
- [x] Changelog.
- [x] Documentation.
- [x] Integration test mapping (using `api/test/integration/mapping/_test_mapping.py`).

# Issue Update

When manually trying to install the package, a build error arises:
<details><summary>Building error</summary>

```console
# /var/ossec/framework/python/bin/pip3 install pyarrow
Collecting pyarrow
  Using cached pyarrow-14.0.1.tar.gz (1.1 MB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Requirement already satisfied: numpy>=1.16.6 in /var/ossec/framework/python/lib/python3.9/site-packages (from pyarrow) (1.19.3)
Building wheels for collected packages: pyarrow
  Building wheel for pyarrow (pyproject.toml) ... error
  error: subprocess-exited-with-error
  
  × Building wheel for pyarrow (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [340 lines of output]
      <string>:34: DeprecationWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html
      running bdist_wheel
      running build
      running build_py
      creating build
      creating build/lib.linux-x86_64-cpython-39
      creating build/lib.linux-x86_64-cpython-39/pyarrow
      copying pyarrow/filesystem.py -> build/lib.linux-x86_64-cpython-39/pyarrow
      copying pyarrow/flight.py -> build/lib.linux-x86_64-cpython-39/pyarrow
      copying pyarrow/compute.py -> build/lib.linux-x86_64-cpython-39/pyarrow
      copying pyarrow/cuda.py -> build/lib.linux-x86_64-cpython-39/pyarrow
      copying pyarrow/orc.py -> build/lib.linux-x86_64-cpython-39/pyarrow
      copying pyarrow/__init__.py -> build/lib.linux-x86_64-cpython-39/pyarrow
      copying pyarrow/dataset.py -> build/lib.linux-x86_64-cpython-39/pyarrow
      copying pyarrow/acero.py -> build/lib.linux-x86_64-cpython-39/pyarrow
      copying pyarrow/fs.py -> build/lib.linux-x86_64-cpython-39/pyarrow
      copying pyarrow/json.py -> build/lib.linux-x86_64-cpython-39/pyarrow
      copying pyarrow/csv.py -> build/lib.linux-x86_64-cpython-39/pyarrow
      copying pyarrow/substrait.py -> build/lib.linux-x86_64-cpython-39/pyarrow
      copying pyarrow/cffi.py -> build/lib.linux-x86_64-cpython-39/pyarrow
      copying pyarrow/types.py -> build/lib.linux-x86_64-cpython-39/pyarrow
      copying pyarrow/pandas_compat.py -> build/lib.linux-x86_64-cpython-39/pyarrow
      copying pyarrow/ipc.py -> build/lib.linux-x86_64-cpython-39/pyarrow
      copying pyarrow/benchmark.py -> build/lib.linux-x86_64-cpython-39/pyarrow
      copying pyarrow/_generated_version.py -> build/lib.linux-x86_64-cpython-39/pyarrow
      copying pyarrow/util.py -> build/lib.linux-x86_64-cpython-39/pyarrow
      copying pyarrow/feather.py -> build/lib.linux-x86_64-cpython-39/pyarrow
      copying pyarrow/jvm.py -> build/lib.linux-x86_64-cpython-39/pyarrow
      copying pyarrow/hdfs.py -> build/lib.linux-x86_64-cpython-39/pyarrow
      copying pyarrow/conftest.py -> build/lib.linux-x86_64-cpython-39/pyarrow
      copying pyarrow/_compute_docstrings.py -> build/lib.linux-x86_64-cpython-39/pyarrow
      creating build/lib.linux-x86_64-cpython-39/pyarrow/interchange
      copying pyarrow/interchange/dataframe.py -> build/lib.linux-x86_64-cpython-39/pyarrow/interchange
      copying pyarrow/interchange/__init__.py -> build/lib.linux-x86_64-cpython-39/pyarrow/interchange
      copying pyarrow/interchange/buffer.py -> build/lib.linux-x86_64-cpython-39/pyarrow/interchange
      copying pyarrow/interchange/from_dataframe.py -> build/lib.linux-x86_64-cpython-39/pyarrow/interchange
      copying pyarrow/interchange/column.py -> build/lib.linux-x86_64-cpython-39/pyarrow/interchange
      creating build/lib.linux-x86_64-cpython-39/pyarrow/parquet
      copying pyarrow/parquet/__init__.py -> build/lib.linux-x86_64-cpython-39/pyarrow/parquet
      copying pyarrow/parquet/core.py -> build/lib.linux-x86_64-cpython-39/pyarrow/parquet
      copying pyarrow/parquet/encryption.py -> build/lib.linux-x86_64-cpython-39/pyarrow/parquet
      creating build/lib.linux-x86_64-cpython-39/pyarrow/tests
      copying pyarrow/tests/test_dataset_encryption.py -> build/lib.linux-x86_64-cpython-39/pyarrow/tests
      copying pyarrow/tests/test_udf.py -> build/lib.linux-x86_64-cpython-39/pyarrow/tests
      copying pyarrow/tests/arrow_16597.py -> build/lib.linux-x86_64-cpython-39/pyarrow/tests
      copying pyarrow/tests/test_jvm.py -> build/lib.linux-x86_64-cpython-39/pyarrow/tests
      copying pyarrow/tests/test_array.py -> build/lib.linux-x86_64-cpython-39/pyarrow/tests
      copying pyarrow/tests/test_misc.py -> build/lib.linux-x86_64-cpython-39/pyarrow/tests
      copying pyarrow/tests/test_hdfs.py -> build/lib.linux-x86_64-cpython-39/pyarrow/tests
      copying pyarrow/tests/test_filesystem.py -> build/lib.linux-x86_64-cpython-39/pyarrow/tests
      copying pyarrow/tests/test_memory.py -> build/lib.linux-x86_64-cpython-39/pyarrow/tests
      copying pyarrow/tests/test_exec_plan.py -> build/lib.linux-x86_64-cpython-39/pyarrow/tests
      copying pyarrow/tests/test_util.py -> build/lib.linux-x86_64-cpython-39/pyarrow/tests
      copying pyarrow/tests/test_gdb.py -> build/lib.linux-x86_64-cpython-39/pyarrow/tests
      copying pyarrow/tests/test_cuda_numba_interop.py -> build/lib.linux-x86_64-cpython-39/pyarrow/tests
      copying pyarrow/tests/test_ipc.py -> build/lib.linux-x86_64-cpython-39/pyarrow/tests
      copying pyarrow/tests/pandas_examples.py -> build/lib.linux-x86_64-cpython-39/pyarrow/tests
      copying pyarrow/tests/test_convert_builtin.py -> build/lib.linux-x86_64-cpython-39/pyarrow/tests
      copying pyarrow/tests/__init__.py -> build/lib.linux-x86_64-cpython-39/pyarrow/tests
      copying pyarrow/tests/test_types.py -> build/lib.linux-x86_64-cpython-39/pyarrow/tests
      copying pyarrow/tests/test_substrait.py -> build/lib.linux-x86_64-cpython-39/pyarrow/tests
      copying pyarrow/tests/test_cython.py -> build/lib.linux-x86_64-cpython-39/pyarrow/tests
      copying pyarrow/tests/test_pandas.py -> build/lib.linux-x86_64-cpython-39/pyarrow/tests
      copying pyarrow/tests/test_compute.py -> build/lib.linux-x86_64-cpython-39/pyarrow/tests
      copying pyarrow/tests/test_cffi.py -> build/lib.linux-x86_64-cpython-39/pyarrow/tests
      copying pyarrow/tests/test_gandiva.py -> build/lib.linux-x86_64-cpython-39/pyarrow/tests
      copying pyarrow/tests/pandas_threaded_import.py -> build/lib.linux-x86_64-cpython-39/pyarrow/tests
      copying pyarrow/tests/test_acero.py -> build/lib.linux-x86_64-cpython-39/pyarrow/tests
      copying pyarrow/tests/test_json.py -> build/lib.linux-x86_64-cpython-39/pyarrow/tests
      copying pyarrow/tests/strategies.py -> build/lib.linux-x86_64-cpython-39/pyarrow/tests
      copying pyarrow/tests/test_scalars.py -> build/lib.linux-x86_64-cpython-39/pyarrow/tests
      copying pyarrow/tests/test_tensor.py -> build/lib.linux-x86_64-cpython-39/pyarrow/tests
      copying pyarrow/tests/test_io.py -> build/lib.linux-x86_64-cpython-39/pyarrow/tests
      copying pyarrow/tests/test_schema.py -> build/lib.linux-x86_64-cpython-39/pyarrow/tests
      copying pyarrow/tests/test_csv.py -> build/lib.linux-x86_64-cpython-39/pyarrow/tests
      copying pyarrow/tests/test_table.py -> build/lib.linux-x86_64-cpython-39/pyarrow/tests
      copying pyarrow/tests/test_adhoc_memory_leak.py -> build/lib.linux-x86_64-cpython-39/pyarrow/tests
      copying pyarrow/tests/test_feather.py -> build/lib.linux-x86_64-cpython-39/pyarrow/tests
      copying pyarrow/tests/test_cuda.py -> build/lib.linux-x86_64-cpython-39/pyarrow/tests
      copying pyarrow/tests/test_fs.py -> build/lib.linux-x86_64-cpython-39/pyarrow/tests
      copying pyarrow/tests/test_flight_async.py -> build/lib.linux-x86_64-cpython-39/pyarrow/tests
      copying pyarrow/tests/util.py -> build/lib.linux-x86_64-cpython-39/pyarrow/tests
      copying pyarrow/tests/test_sparse_tensor.py -> build/lib.linux-x86_64-cpython-39/pyarrow/tests
      copying pyarrow/tests/test_builder.py -> build/lib.linux-x86_64-cpython-39/pyarrow/tests
      copying pyarrow/tests/test_extension_type.py -> build/lib.linux-x86_64-cpython-39/pyarrow/tests
      copying pyarrow/tests/test_cpp_internals.py -> build/lib.linux-x86_64-cpython-39/pyarrow/tests
      copying pyarrow/tests/test_dataset.py -> build/lib.linux-x86_64-cpython-39/pyarrow/tests
      copying pyarrow/tests/test_deprecations.py -> build/lib.linux-x86_64-cpython-39/pyarrow/tests
      copying pyarrow/tests/test_strategies.py -> build/lib.linux-x86_64-cpython-39/pyarrow/tests
      copying pyarrow/tests/read_record_batch.py -> build/lib.linux-x86_64-cpython-39/pyarrow/tests
      copying pyarrow/tests/test_orc.py -> build/lib.linux-x86_64-cpython-39/pyarrow/tests
      copying pyarrow/tests/conftest.py -> build/lib.linux-x86_64-cpython-39/pyarrow/tests
      copying pyarrow/tests/arrow_7980.py -> build/lib.linux-x86_64-cpython-39/pyarrow/tests
      copying pyarrow/tests/test_flight.py -> build/lib.linux-x86_64-cpython-39/pyarrow/tests
      creating build/lib.linux-x86_64-cpython-39/pyarrow/vendored
      copying pyarrow/vendored/docscrape.py -> build/lib.linux-x86_64-cpython-39/pyarrow/vendored
      copying pyarrow/vendored/version.py -> build/lib.linux-x86_64-cpython-39/pyarrow/vendored
      copying pyarrow/vendored/__init__.py -> build/lib.linux-x86_64-cpython-39/pyarrow/vendored
      creating build/lib.linux-x86_64-cpython-39/pyarrow/tests/interchange
      copying pyarrow/tests/interchange/__init__.py -> build/lib.linux-x86_64-cpython-39/pyarrow/tests/interchange
      copying pyarrow/tests/interchange/test_interchange_spec.py -> build/lib.linux-x86_64-cpython-39/pyarrow/tests/interchange
      copying pyarrow/tests/interchange/test_conversion.py -> build/lib.linux-x86_64-cpython-39/pyarrow/tests/interchange
      creating build/lib.linux-x86_64-cpython-39/pyarrow/tests/parquet
      copying pyarrow/tests/parquet/test_parquet_file.py -> build/lib.linux-x86_64-cpython-39/pyarrow/tests/parquet
      copying pyarrow/tests/parquet/test_encryption.py -> build/lib.linux-x86_64-cpython-39/pyarrow/tests/parquet
      copying pyarrow/tests/parquet/__init__.py -> build/lib.linux-x86_64-cpython-39/pyarrow/tests/parquet
      copying pyarrow/tests/parquet/test_pandas.py -> build/lib.linux-x86_64-cpython-39/pyarrow/tests/parquet
      copying pyarrow/tests/parquet/common.py -> build/lib.linux-x86_64-cpython-39/pyarrow/tests/parquet
      copying pyarrow/tests/parquet/test_parquet_writer.py -> build/lib.linux-x86_64-cpython-39/pyarrow/tests/parquet
      copying pyarrow/tests/parquet/test_basic.py -> build/lib.linux-x86_64-cpython-39/pyarrow/tests/parquet
      copying pyarrow/tests/parquet/test_metadata.py -> build/lib.linux-x86_64-cpython-39/pyarrow/tests/parquet
      copying pyarrow/tests/parquet/test_compliant_nested_type.py -> build/lib.linux-x86_64-cpython-39/pyarrow/tests/parquet
      copying pyarrow/tests/parquet/test_datetime.py -> build/lib.linux-x86_64-cpython-39/pyarrow/tests/parquet
      copying pyarrow/tests/parquet/test_dataset.py -> build/lib.linux-x86_64-cpython-39/pyarrow/tests/parquet
      copying pyarrow/tests/parquet/encryption.py -> build/lib.linux-x86_64-cpython-39/pyarrow/tests/parquet
      copying pyarrow/tests/parquet/conftest.py -> build/lib.linux-x86_64-cpython-39/pyarrow/tests/parquet
      copying pyarrow/tests/parquet/test_data_types.py -> build/lib.linux-x86_64-cpython-39/pyarrow/tests/parquet
      running egg_info
      writing pyarrow.egg-info/PKG-INFO
      writing dependency_links to pyarrow.egg-info/dependency_links.txt
      writing requirements to pyarrow.egg-info/requires.txt
      writing top-level names to pyarrow.egg-info/top_level.txt
      listing git files failed - pretending there aren't any
      reading manifest file 'pyarrow.egg-info/SOURCES.txt'
      reading manifest template 'MANIFEST.in'
      warning: no files found matching '../LICENSE.txt'
      warning: no files found matching '../NOTICE.txt'
      warning: no previously-included files matching '*.so' found anywhere in distribution
      warning: no previously-included files matching '*.pyc' found anywhere in distribution
      warning: no previously-included files matching '*~' found anywhere in distribution
      warning: no previously-included files matching '#*' found anywhere in distribution
      warning: no previously-included files matching '.git*' found anywhere in distribution
      warning: no previously-included files matching '.DS_Store' found anywhere in distribution
      no previously-included directories found matching '.asv'
      writing manifest file 'pyarrow.egg-info/SOURCES.txt'
      copying pyarrow/__init__.pxd -> build/lib.linux-x86_64-cpython-39/pyarrow
      copying pyarrow/_acero.pxd -> build/lib.linux-x86_64-cpython-39/pyarrow
      copying pyarrow/_acero.pyx -> build/lib.linux-x86_64-cpython-39/pyarrow
      copying pyarrow/_compute.pxd -> build/lib.linux-x86_64-cpython-39/pyarrow
      copying pyarrow/_compute.pyx -> build/lib.linux-x86_64-cpython-39/pyarrow
      copying pyarrow/_csv.pxd -> build/lib.linux-x86_64-cpython-39/pyarrow
      copying pyarrow/_csv.pyx -> build/lib.linux-x86_64-cpython-39/pyarrow
      copying pyarrow/_cuda.pxd -> build/lib.linux-x86_64-cpython-39/pyarrow
      copying pyarrow/_cuda.pyx -> build/lib.linux-x86_64-cpython-39/pyarrow
      copying pyarrow/_dataset.pxd -> build/lib.linux-x86_64-cpython-39/pyarrow
      copying pyarrow/_dataset.pyx -> build/lib.linux-x86_64-cpython-39/pyarrow
      copying pyarrow/_dataset_orc.pyx -> build/lib.linux-x86_64-cpython-39/pyarrow
      copying pyarrow/_dataset_parquet.pxd -> build/lib.linux-x86_64-cpython-39/pyarrow
      copying pyarrow/_dataset_parquet.pyx -> build/lib.linux-x86_64-cpython-39/pyarrow
      copying pyarrow/_dataset_parquet_encryption.pyx -> build/lib.linux-x86_64-cpython-39/pyarrow
      copying pyarrow/_feather.pyx -> build/lib.linux-x86_64-cpython-39/pyarrow
      copying pyarrow/_flight.pyx -> build/lib.linux-x86_64-cpython-39/pyarrow
      copying pyarrow/_fs.pxd -> build/lib.linux-x86_64-cpython-39/pyarrow
      copying pyarrow/_fs.pyx -> build/lib.linux-x86_64-cpython-39/pyarrow
      copying pyarrow/_gcsfs.pyx -> build/lib.linux-x86_64-cpython-39/pyarrow
      copying pyarrow/_hdfs.pyx -> build/lib.linux-x86_64-cpython-39/pyarrow
      copying pyarrow/_hdfsio.pyx -> build/lib.linux-x86_64-cpython-39/pyarrow
      copying pyarrow/_json.pxd -> build/lib.linux-x86_64-cpython-39/pyarrow
      copying pyarrow/_json.pyx -> build/lib.linux-x86_64-cpython-39/pyarrow
      copying pyarrow/_orc.pxd -> build/lib.linux-x86_64-cpython-39/pyarrow
      copying pyarrow/_orc.pyx -> build/lib.linux-x86_64-cpython-39/pyarrow
      copying pyarrow/_parquet.pxd -> build/lib.linux-x86_64-cpython-39/pyarrow
      copying pyarrow/_parquet.pyx -> build/lib.linux-x86_64-cpython-39/pyarrow
      copying pyarrow/_parquet_encryption.pxd -> build/lib.linux-x86_64-cpython-39/pyarrow
      copying pyarrow/_parquet_encryption.pyx -> build/lib.linux-x86_64-cpython-39/pyarrow
      copying pyarrow/_pyarrow_cpp_tests.pxd -> build/lib.linux-x86_64-cpython-39/pyarrow
      copying pyarrow/_pyarrow_cpp_tests.pyx -> build/lib.linux-x86_64-cpython-39/pyarrow
      copying pyarrow/_s3fs.pyx -> build/lib.linux-x86_64-cpython-39/pyarrow
      copying pyarrow/_substrait.pyx -> build/lib.linux-x86_64-cpython-39/pyarrow
      copying pyarrow/array.pxi -> build/lib.linux-x86_64-cpython-39/pyarrow
      copying pyarrow/benchmark.pxi -> build/lib.linux-x86_64-cpython-39/pyarrow
      copying pyarrow/builder.pxi -> build/lib.linux-x86_64-cpython-39/pyarrow
      copying pyarrow/compat.pxi -> build/lib.linux-x86_64-cpython-39/pyarrow
      copying pyarrow/config.pxi -> build/lib.linux-x86_64-cpython-39/pyarrow
      copying pyarrow/error.pxi -> build/lib.linux-x86_64-cpython-39/pyarrow
      copying pyarrow/gandiva.pyx -> build/lib.linux-x86_64-cpython-39/pyarrow
      copying pyarrow/io.pxi -> build/lib.linux-x86_64-cpython-39/pyarrow
      copying pyarrow/ipc.pxi -> build/lib.linux-x86_64-cpython-39/pyarrow
      copying pyarrow/lib.pxd -> build/lib.linux-x86_64-cpython-39/pyarrow
      copying pyarrow/lib.pyx -> build/lib.linux-x86_64-cpython-39/pyarrow
      copying pyarrow/memory.pxi -> build/lib.linux-x86_64-cpython-39/pyarrow
      copying pyarrow/pandas-shim.pxi -> build/lib.linux-x86_64-cpython-39/pyarrow
      copying pyarrow/public-api.pxi -> build/lib.linux-x86_64-cpython-39/pyarrow
      copying pyarrow/scalar.pxi -> build/lib.linux-x86_64-cpython-39/pyarrow
      copying pyarrow/table.pxi -> build/lib.linux-x86_64-cpython-39/pyarrow
      copying pyarrow/tensor.pxi -> build/lib.linux-x86_64-cpython-39/pyarrow
      copying pyarrow/types.pxi -> build/lib.linux-x86_64-cpython-39/pyarrow
      creating build/lib.linux-x86_64-cpython-39/pyarrow/includes
      copying pyarrow/includes/libarrow_feather.pxd -> build/lib.linux-x86_64-cpython-39/pyarrow/includes
      copying pyarrow/includes/libarrow.pxd -> build/lib.linux-x86_64-cpython-39/pyarrow/includes
      copying pyarrow/includes/libarrow_python.pxd -> build/lib.linux-x86_64-cpython-39/pyarrow/includes
      copying pyarrow/includes/libarrow_flight.pxd -> build/lib.linux-x86_64-cpython-39/pyarrow/includes
      copying pyarrow/includes/libgandiva.pxd -> build/lib.linux-x86_64-cpython-39/pyarrow/includes
      copying pyarrow/includes/libarrow_dataset_parquet.pxd -> build/lib.linux-x86_64-cpython-39/pyarrow/includes
      copying pyarrow/includes/libarrow_fs.pxd -> build/lib.linux-x86_64-cpython-39/pyarrow/includes
      copying pyarrow/includes/libarrow_cuda.pxd -> build/lib.linux-x86_64-cpython-39/pyarrow/includes
      copying pyarrow/includes/libarrow_substrait.pxd -> build/lib.linux-x86_64-cpython-39/pyarrow/includes
      copying pyarrow/includes/__init__.pxd -> build/lib.linux-x86_64-cpython-39/pyarrow/includes
      copying pyarrow/includes/libarrow_dataset.pxd -> build/lib.linux-x86_64-cpython-39/pyarrow/includes
      copying pyarrow/includes/libparquet_encryption.pxd -> build/lib.linux-x86_64-cpython-39/pyarrow/includes
      copying pyarrow/includes/common.pxd -> build/lib.linux-x86_64-cpython-39/pyarrow/includes
      copying pyarrow/includes/libarrow_acero.pxd -> build/lib.linux-x86_64-cpython-39/pyarrow/includes
      copying pyarrow/tests/bound_function_visit_strings.pyx -> build/lib.linux-x86_64-cpython-39/pyarrow/tests
      copying pyarrow/tests/extensions.pyx -> build/lib.linux-x86_64-cpython-39/pyarrow/tests
      copying pyarrow/tests/pyarrow_cython_example.pyx -> build/lib.linux-x86_64-cpython-39/pyarrow/tests
      creating build/lib.linux-x86_64-cpython-39/pyarrow/src
      creating build/lib.linux-x86_64-cpython-39/pyarrow/src/arrow
      creating build/lib.linux-x86_64-cpython-39/pyarrow/src/arrow/python
      copying pyarrow/src/arrow/python/CMakeLists.txt -> build/lib.linux-x86_64-cpython-39/pyarrow/src/arrow/python
      copying pyarrow/src/arrow/python/api.h -> build/lib.linux-x86_64-cpython-39/pyarrow/src/arrow/python
      copying pyarrow/src/arrow/python/arrow_to_pandas.cc -> build/lib.linux-x86_64-cpython-39/pyarrow/src/arrow/python
      copying pyarrow/src/arrow/python/arrow_to_pandas.h -> build/lib.linux-x86_64-cpython-39/pyarrow/src/arrow/python
      copying pyarrow/src/arrow/python/arrow_to_python_internal.h -> build/lib.linux-x86_64-cpython-39/pyarrow/src/arrow/python
      copying pyarrow/src/arrow/python/async.h -> build/lib.linux-x86_64-cpython-39/pyarrow/src/arrow/python
      copying pyarrow/src/arrow/python/benchmark.cc -> build/lib.linux-x86_64-cpython-39/pyarrow/src/arrow/python
      copying pyarrow/src/arrow/python/benchmark.h -> build/lib.linux-x86_64-cpython-39/pyarrow/src/arrow/python
      copying pyarrow/src/arrow/python/common.cc -> build/lib.linux-x86_64-cpython-39/pyarrow/src/arrow/python
      copying pyarrow/src/arrow/python/common.h -> build/lib.linux-x86_64-cpython-39/pyarrow/src/arrow/python
      copying pyarrow/src/arrow/python/csv.cc -> build/lib.linux-x86_64-cpython-39/pyarrow/src/arrow/python
      copying pyarrow/src/arrow/python/csv.h -> build/lib.linux-x86_64-cpython-39/pyarrow/src/arrow/python
      copying pyarrow/src/arrow/python/datetime.cc -> build/lib.linux-x86_64-cpython-39/pyarrow/src/arrow/python
      copying pyarrow/src/arrow/python/datetime.h -> build/lib.linux-x86_64-cpython-39/pyarrow/src/arrow/python
      copying pyarrow/src/arrow/python/decimal.cc -> build/lib.linux-x86_64-cpython-39/pyarrow/src/arrow/python
      copying pyarrow/src/arrow/python/decimal.h -> build/lib.linux-x86_64-cpython-39/pyarrow/src/arrow/python
      copying pyarrow/src/arrow/python/deserialize.cc -> build/lib.linux-x86_64-cpython-39/pyarrow/src/arrow/python
      copying pyarrow/src/arrow/python/deserialize.h -> build/lib.linux-x86_64-cpython-39/pyarrow/src/arrow/python
      copying pyarrow/src/arrow/python/extension_type.cc -> build/lib.linux-x86_64-cpython-39/pyarrow/src/arrow/python
      copying pyarrow/src/arrow/python/extension_type.h -> build/lib.linux-x86_64-cpython-39/pyarrow/src/arrow/python
      copying pyarrow/src/arrow/python/filesystem.cc -> build/lib.linux-x86_64-cpython-39/pyarrow/src/arrow/python
      copying pyarrow/src/arrow/python/filesystem.h -> build/lib.linux-x86_64-cpython-39/pyarrow/src/arrow/python
      copying pyarrow/src/arrow/python/flight.cc -> build/lib.linux-x86_64-cpython-39/pyarrow/src/arrow/python
      copying pyarrow/src/arrow/python/flight.h -> build/lib.linux-x86_64-cpython-39/pyarrow/src/arrow/python
      copying pyarrow/src/arrow/python/gdb.cc -> build/lib.linux-x86_64-cpython-39/pyarrow/src/arrow/python
      copying pyarrow/src/arrow/python/gdb.h -> build/lib.linux-x86_64-cpython-39/pyarrow/src/arrow/python
      copying pyarrow/src/arrow/python/helpers.cc -> build/lib.linux-x86_64-cpython-39/pyarrow/src/arrow/python
      copying pyarrow/src/arrow/python/helpers.h -> build/lib.linux-x86_64-cpython-39/pyarrow/src/arrow/python
      copying pyarrow/src/arrow/python/inference.cc -> build/lib.linux-x86_64-cpython-39/pyarrow/src/arrow/python
      copying pyarrow/src/arrow/python/inference.h -> build/lib.linux-x86_64-cpython-39/pyarrow/src/arrow/python
      copying pyarrow/src/arrow/python/init.cc -> build/lib.linux-x86_64-cpython-39/pyarrow/src/arrow/python
      copying pyarrow/src/arrow/python/init.h -> build/lib.linux-x86_64-cpython-39/pyarrow/src/arrow/python
      copying pyarrow/src/arrow/python/io.cc -> build/lib.linux-x86_64-cpython-39/pyarrow/src/arrow/python
      copying pyarrow/src/arrow/python/io.h -> build/lib.linux-x86_64-cpython-39/pyarrow/src/arrow/python
      copying pyarrow/src/arrow/python/ipc.cc -> build/lib.linux-x86_64-cpython-39/pyarrow/src/arrow/python
      copying pyarrow/src/arrow/python/ipc.h -> build/lib.linux-x86_64-cpython-39/pyarrow/src/arrow/python
      copying pyarrow/src/arrow/python/iterators.h -> build/lib.linux-x86_64-cpython-39/pyarrow/src/arrow/python
      copying pyarrow/src/arrow/python/numpy_convert.cc -> build/lib.linux-x86_64-cpython-39/pyarrow/src/arrow/python
      copying pyarrow/src/arrow/python/numpy_convert.h -> build/lib.linux-x86_64-cpython-39/pyarrow/src/arrow/python
      copying pyarrow/src/arrow/python/numpy_internal.h -> build/lib.linux-x86_64-cpython-39/pyarrow/src/arrow/python
      copying pyarrow/src/arrow/python/numpy_interop.h -> build/lib.linux-x86_64-cpython-39/pyarrow/src/arrow/python
      copying pyarrow/src/arrow/python/numpy_to_arrow.cc -> build/lib.linux-x86_64-cpython-39/pyarrow/src/arrow/python
      copying pyarrow/src/arrow/python/numpy_to_arrow.h -> build/lib.linux-x86_64-cpython-39/pyarrow/src/arrow/python
      copying pyarrow/src/arrow/python/parquet_encryption.cc -> build/lib.linux-x86_64-cpython-39/pyarrow/src/arrow/python
      copying pyarrow/src/arrow/python/parquet_encryption.h -> build/lib.linux-x86_64-cpython-39/pyarrow/src/arrow/python
      copying pyarrow/src/arrow/python/pch.h -> build/lib.linux-x86_64-cpython-39/pyarrow/src/arrow/python
      copying pyarrow/src/arrow/python/platform.h -> build/lib.linux-x86_64-cpython-39/pyarrow/src/arrow/python
      copying pyarrow/src/arrow/python/pyarrow.cc -> build/lib.linux-x86_64-cpython-39/pyarrow/src/arrow/python
      copying pyarrow/src/arrow/python/pyarrow.h -> build/lib.linux-x86_64-cpython-39/pyarrow/src/arrow/python
      copying pyarrow/src/arrow/python/pyarrow_api.h -> build/lib.linux-x86_64-cpython-39/pyarrow/src/arrow/python
      copying pyarrow/src/arrow/python/pyarrow_lib.h -> build/lib.linux-x86_64-cpython-39/pyarrow/src/arrow/python
      copying pyarrow/src/arrow/python/python_test.cc -> build/lib.linux-x86_64-cpython-39/pyarrow/src/arrow/python
      copying pyarrow/src/arrow/python/python_test.h -> build/lib.linux-x86_64-cpython-39/pyarrow/src/arrow/python
      copying pyarrow/src/arrow/python/python_to_arrow.cc -> build/lib.linux-x86_64-cpython-39/pyarrow/src/arrow/python
      copying pyarrow/src/arrow/python/python_to_arrow.h -> build/lib.linux-x86_64-cpython-39/pyarrow/src/arrow/python
      copying pyarrow/src/arrow/python/serialize.cc -> build/lib.linux-x86_64-cpython-39/pyarrow/src/arrow/python
      copying pyarrow/src/arrow/python/serialize.h -> build/lib.linux-x86_64-cpython-39/pyarrow/src/arrow/python
      copying pyarrow/src/arrow/python/type_traits.h -> build/lib.linux-x86_64-cpython-39/pyarrow/src/arrow/python
      copying pyarrow/src/arrow/python/udf.cc -> build/lib.linux-x86_64-cpython-39/pyarrow/src/arrow/python
      copying pyarrow/src/arrow/python/udf.h -> build/lib.linux-x86_64-cpython-39/pyarrow/src/arrow/python
      copying pyarrow/src/arrow/python/visibility.h -> build/lib.linux-x86_64-cpython-39/pyarrow/src/arrow/python
      creating build/lib.linux-x86_64-cpython-39/pyarrow/tests/data
      creating build/lib.linux-x86_64-cpython-39/pyarrow/tests/data/feather
      copying pyarrow/tests/data/feather/v0.17.0.version.2-compression.lz4.feather -> build/lib.linux-x86_64-cpython-39/pyarrow/tests/data/feather
      creating build/lib.linux-x86_64-cpython-39/pyarrow/tests/data/orc
      copying pyarrow/tests/data/orc/README.md -> build/lib.linux-x86_64-cpython-39/pyarrow/tests/data/orc
      copying pyarrow/tests/data/orc/TestOrcFile.emptyFile.jsn.gz -> build/lib.linux-x86_64-cpython-39/pyarrow/tests/data/orc
      copying pyarrow/tests/data/orc/TestOrcFile.emptyFile.orc -> build/lib.linux-x86_64-cpython-39/pyarrow/tests/data/orc
      copying pyarrow/tests/data/orc/TestOrcFile.test1.jsn.gz -> build/lib.linux-x86_64-cpython-39/pyarrow/tests/data/orc
      copying pyarrow/tests/data/orc/TestOrcFile.test1.orc -> build/lib.linux-x86_64-cpython-39/pyarrow/tests/data/orc
      copying pyarrow/tests/data/orc/TestOrcFile.testDate1900.jsn.gz -> build/lib.linux-x86_64-cpython-39/pyarrow/tests/data/orc
      copying pyarrow/tests/data/orc/TestOrcFile.testDate1900.orc -> build/lib.linux-x86_64-cpython-39/pyarrow/tests/data/orc
      copying pyarrow/tests/data/orc/decimal.jsn.gz -> build/lib.linux-x86_64-cpython-39/pyarrow/tests/data/orc
      copying pyarrow/tests/data/orc/decimal.orc -> build/lib.linux-x86_64-cpython-39/pyarrow/tests/data/orc
      creating build/lib.linux-x86_64-cpython-39/pyarrow/tests/data/parquet
      copying pyarrow/tests/data/parquet/v0.7.1.all-named-index.parquet -> build/lib.linux-x86_64-cpython-39/pyarrow/tests/data/parquet
      copying pyarrow/tests/data/parquet/v0.7.1.column-metadata-handling.parquet -> build/lib.linux-x86_64-cpython-39/pyarrow/tests/data/parquet
      copying pyarrow/tests/data/parquet/v0.7.1.parquet -> build/lib.linux-x86_64-cpython-39/pyarrow/tests/data/parquet
      copying pyarrow/tests/data/parquet/v0.7.1.some-named-index.parquet -> build/lib.linux-x86_64-cpython-39/pyarrow/tests/data/parquet
      running build_ext
      creating /tmp/pip-install-xrj6j0st/pyarrow_46d8841eaf594d0abc12e467ee56da16/build/temp.linux-x86_64-cpython-39
      -- Running cmake for PyArrow
      cmake -DCMAKE_INSTALL_PREFIX=/tmp/pip-install-xrj6j0st/pyarrow_46d8841eaf594d0abc12e467ee56da16/build/lib.linux-x86_64-cpython-39/pyarrow -DPYTHON_EXECUTABLE=/var/ossec/framework/python/bin/python3.9 -DPython3_EXECUTABLE=/var/ossec/framework/python/bin/python3.9 -DPYARROW_CXXFLAGS= -DPYARROW_BUILD_CUDA=off -DPYARROW_BUILD_SUBSTRAIT=off -DPYARROW_BUILD_FLIGHT=off -DPYARROW_BUILD_GANDIVA=off -DPYARROW_BUILD_ACERO=off -DPYARROW_BUILD_DATASET=off -DPYARROW_BUILD_ORC=off -DPYARROW_BUILD_PARQUET=off -DPYARROW_BUILD_PARQUET_ENCRYPTION=off -DPYARROW_BUILD_GCS=off -DPYARROW_BUILD_S3=off -DPYARROW_BUILD_HDFS=off -DPYARROW_BUNDLE_ARROW_CPP=off -DPYARROW_BUNDLE_CYTHON_CPP=off -DPYARROW_GENERATE_COVERAGE=off -DCMAKE_BUILD_TYPE=release /tmp/pip-install-xrj6j0st/pyarrow_46d8841eaf594d0abc12e467ee56da16
      -- The C compiler identification is GNU 9.4.0
      -- The CXX compiler identification is GNU 9.4.0
      -- Detecting C compiler ABI info
      -- Detecting C compiler ABI info - done
      -- Check for working C compiler: /usr/bin/cc - skipped
      -- Detecting C compile features
      -- Detecting C compile features - done
      -- Detecting CXX compiler ABI info
      -- Detecting CXX compiler ABI info - done
      -- Check for working CXX compiler: /usr/bin/c++ - skipped
      -- Detecting CXX compile features
      -- Detecting CXX compile features - done
      -- System processor: x86_64
      -- Performing Test CXX_SUPPORTS_SSE4_2
      -- Performing Test CXX_SUPPORTS_SSE4_2 - Success
      -- Performing Test CXX_SUPPORTS_AVX2
      -- Performing Test CXX_SUPPORTS_AVX2 - Success
      -- Performing Test CXX_SUPPORTS_AVX512
      -- Performing Test CXX_SUPPORTS_AVX512 - Failed
      -- Arrow build warning level: PRODUCTION
      -- Using ld linker
      -- Build Type: RELEASE
      -- CMAKE_C_FLAGS:  -Wall -fno-semantic-interposition -msse4.2  -fdiagnostics-color=always  -fno-omit-frame-pointer -Wno-unused-variable -Wno-maybe-uninitialized
      -- CMAKE_CXX_FLAGS:  -Wno-noexcept-type  -Wall -fno-semantic-interposition -msse4.2  -fdiagnostics-color=always  -fno-omit-frame-pointer -Wno-unused-variable -Wno-maybe-uninitialized
      -- Generator: Unix Makefiles
      -- Build output directory: /tmp/pip-install-xrj6j0st/pyarrow_46d8841eaf594d0abc12e467ee56da16/build/temp.linux-x86_64-cpython-39/release
      -- Found Python3: /var/ossec/framework/python/bin/python3.9 (found version "3.9.18") found components: Interpreter Development.Module NumPy
      -- Found Python3Alt: /var/ossec/framework/python/bin/python3.9
      CMake Error at CMakeLists.txt:268 (find_package):
        By not providing "FindArrow.cmake" in CMAKE_MODULE_PATH this project has
        asked CMake to find a package configuration file provided by "Arrow", but
        CMake did not find one.
      
        Could not find a package configuration file provided by "Arrow" with any of
        the following names:
      
          ArrowConfig.cmake
          arrow-config.cmake
      
        Add the installation prefix of "Arrow" to CMAKE_PREFIX_PATH or set
        "Arrow_DIR" to a directory containing one of the above files.  If "Arrow"
        provides a separate development package or SDK, be sure it has been
        installed.
      
      
      -- Configuring incomplete, errors occurred!
      See also "/tmp/pip-install-xrj6j0st/pyarrow_46d8841eaf594d0abc12e467ee56da16/build/temp.linux-x86_64-cpython-39/CMakeFiles/CMakeOutput.log".
      See also "/tmp/pip-install-xrj6j0st/pyarrow_46d8841eaf594d0abc12e467ee56da16/build/temp.linux-x86_64-cpython-39/CMakeFiles/CMakeError.log".
      error: command '/usr/local/bin/cmake' failed with exit code 1
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for pyarrow
Failed to build pyarrow
ERROR: Could not build wheels for pyarrow, which is required to install pyproject.toml-based projects
```

</details>

--- a/framework/requirements.txt
+++ b/framework/requirements.txt
@@ -62,6 +62,7 @@ protobuf==3.19.6
 proto-plus==1.19.0
 psutil==5.8.0
 pyarrow==8.0.0
+pyarrow_hotfix==0.5
 pyasn1==0.4.8
 pyasn1-modules==0.2.8
 pycparser==2.20

--- a/wodles/aws/aws_s3.py
+++ b/wodles/aws/aws_s3.py
@@ -47,6 +47,7 @@ except ImportError:
 
 try:
     import pyarrow.parquet as pq
+    import pyarrow_hotfix
 except ImportError:
     print('ERROR: pyarrow module is required.')
     sys.exit(10)

